recipes/full_finetune_single_device.py

```diff
@@ -267,6 +267,13 @@ def setup(self, cfg: DictConfig) -> None:
             self._steps_per_epoch = self.max_steps_per_epoch
         self.global_step = self.epochs_run * self._steps_per_epoch
 
+        # Setup lr scheduler
+        self._lr_scheduler = self._setup_lr_scheduler(
+            cfg_lr_scheduler=cfg.lr_scheduler,
```

Comment:
I might be wrong but I think this will throw an attribute error if LR scheduler is not passed, should probably do something like this instead

```suggestion
            cfg_lr_scheduler=cfg.get("lr_scheduler", None),
```